### PR TITLE
Reduce kubectl error output of secret file

### PR DESCRIPTION
**Pull request by yunzhu-li**
This is https://github.com/NYTimes/drone-gke/pull/61 against `develop` branch. [Test build](https://drone.dv.nyt.net/nytm/dv-yunzhu-sandbox/51/3)

### CODE REVIEW

**Comment by tonglil** [comment]
After https://github.com/NYTimes/drone-gke/pull/65 is merged, can you add to the CHANGELOG notes for `0.8.1`?

**Comment by yunzhu-li** [comment]
Sure

File: main.go [hunk 1]
```diff
@@ -423,6 +445,16 @@ func run(c *cli.Context) error {
 		return fmt.Errorf("Error: %s\n", err)
 	}
 
+	// Apply Kubernetes secrets manifests
+
+	argsSecret := applyArgs(c.Bool("dry-run"), manifestsSecret)
```
Comment (tonglil) [inline main.go]: There is no check in case the Secret manifest is not included (it is an optional field/file).

Comment (yunzhu-li) [reply main.go]: Check added

File: main.go [hunk 2]
```diff
@@ -423,6 +447,18 @@ func run(c *cli.Context) error {
 		return fmt.Errorf("Error: %s\n", err)
 	}
 
+	// Apply Kubernetes secrets manifests
+
+	if len(manifestsSecret) > 0 {
```
Comment (tonglil) [inline main.go]: I think we can improve the logging and tell the user when there is no Secret manifest configured.

Comment (yunzhu-li) [reply main.go]: [This line](https://github.com/NYTimes/drone-gke/pull/66/files#diff-7ddfb3e035b42cd70649cc33393fe32cR333) will print when there's no no Secret manifest. or do we need more / different logging?

Comment (tonglil) [reply main.go]: You are right, that looks good to me.

File: main.go [hunk 3]
```diff
@@ -488,6 +524,16 @@ func applyArgs(dryrun bool, file string) []string {
 	return args
 }
 
+// printTrimmedError prints the last line of stderrbuf to stderr
+func printTrimmedError(stderrbuf bytes.Buffer) {
+	var lastLine string
+	scanner := bufio.NewScanner(strings.NewReader(string(stderrbuf.Bytes())))
```
Comment (jprobinson) [inline main.go]: `bufio.NewScanner` accepts an `io.Reader`, which is what `*bytes.Buffer` implements.

You can simplify this to `bufio.NewScanner(&stderrbuff)`

**Comment by tonglil** [comment]
Waiting for 2 approvals on this.

**Pull request merged**